packages/generator/tests/acceptance/test_streaming_measurement_client.py

```diff
@@ -49,6 +50,61 @@ def test___measurement_plugin_client___stream_measure___returns_output(
     assert responses == expected_output
 
 
+def test___measurement_plugin_client___invoke_measure_from_two_threads___initiates_first_measure_and_rejects_second_measure(
+    measurement_plugin_client_module: ModuleType,
+) -> None:
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+
+    with pytest.raises(RuntimeError) as exc_info:
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
+            future_measure_1 = executor.submit(measurement_plugin_client.measure)
+            future_measure_2 = executor.submit(measurement_plugin_client.measure)
+            future_measure_1.result()
+            future_measure_2.result()
+
+    assert "A measure call is already in progress." in exc_info.value.args[0]
+
+
+def test___non_streaming_measurement_execution___cancel___returns_true(
```

Comment:
The "then" phrase of the test name should emphasize the outcome of the action. Here, "returns true" states the value and doesn't indicate the outcome.
```suggestion
def test___non_streaming_measurement_execution___cancel___cancels_measurement(
```